lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -238,8 +240,35 @@ MuxOp::inferReturnTypes(MLIRContext *context, std::optional<Location> location,
   // operand)
   if (operands.size() < 2)
     return failure();
-  // Result type is type of any data operand
-  inferredReturnTypes.push_back(operands[1].getType());
+
+  // Map to collect extra signals from data operands
+  llvm::DenseMap<StringRef, const ExtraSignal *> extraSignalsMap;
```

Comment:
I think for consistancy it should be unionOfExtraSignals here also?

Reply:
maybe specify in the comments that the map is for verification and the vector is for inferrence.

Question: should we do be doing verification here, or should we just infer the type?